### DISCUSSION TITLE
Action - subtract on (internal) column

### DISCUSSION THREAD

**Discussion by KristofferBroberg** [Budibase/budibase#2744] [Help] [answered]
Hi all!

Excellent product. Will recommend Budibase to everyone I meet, no doubt. One very basic question: how would I set up my action in order to perform simple math on internal number column? 

There are no options for actions on columns, but this very basic operation must be possible, right?

**Comment by joebudi** [comment, marked as answer]
Hey.  

There's 2 possible solutions:
1. Add a formula data type and add the calc
2. Use handlebars (in binding drawer) - you can search the helpers

I hope that helps.
Joe

**Reply by KristofferBroberg** [reply]
Thank you. However, I use a formula and in the view of the table I can't select that column for the calc. I´ve settled on using an external datasource instead.

**Reply by NeloReis** [reply]
Can someone please provide some written examples of math using handlebars in the binding drawer? the binding drawer is a mystery wrapped in an enigma
and all the handlebars sites provide no applicable examples. I'd love to see a couple of examples in the context of budibase .. Rows ??? what the heck is that dataProviderRows ?? sorry .. 16 hours digging to understand rows. in that context. and nothing..  just a couple of examples of math using columns and rows .. I would be on my way.. thank you.

**Reply by mjashanks** [reply]
Hi @NeloReis . Handlebars is not easy for Math at all. This is why we are adding javascript support, but won't be released until the end of next month.

In the meantime, here are a few simple math expressions:

`{{ add 2 4 }}` > result = 6

`{{ add Repeater.Invoices.TotalExTax Repeater.Invoices.TaxAmount }}`  > same example, using fields, repeater row

`{{ multiply Repeater.Invoices.TotalExTax 0.2 }}` > Field Invoices.TotalExVat * 0.2

combining function `{{ multiply (add 2 3) 4 }}` > `(2 + 3) * 4 = 20`


And `Data Provider.Rows` is the number of rows returned by your data provider (which is used to connect your front end to your data sources).

We definitely need more docs. It will be a big focus over the next few months - please feel free to continue to ask questions in here :)

**Reply by NeloReis** [reply]
thank you so very much. I'm still having a hard time understanding how to move data around how to do the math on specific entries.. but i believe your examples will help me. I'm trying to build complex mathematical analysis ... but i can't get if the ground if i cant just add 2 specific entries. and get a result on a card.
Im fighting hard to build something with this project. cause i really believe in it. (I've tested them all ) 
budibase is still my favorite.. 
I've sent 40 year doing the VAR x = (a+b) ..its precise.. i know what is in A and what is in B... but in budibase i don't know how to call A or B or how to store or display X...
but i won't quit...
i still see great potential in this project. really want to learn how to use it.
so thank you sooooo very much.
